execute/tokendata/processor.go

```diff
@@ -0,0 +1,64 @@
+package tokendata
+
+import (
+	"context"
+
+	cciptypes "github.com/smartcontractkit/chainlink-common/pkg/types/ccipocr3"
+
+	"github.com/smartcontractkit/chainlink-ccip/execute/exectypes"
+)
+
+type TokenDataProcessor interface {
```

Comment:
naming nit: we've been using the term "processor" (in this codebase) to reference an object that implements the processor interface defined here: https://github.com/smartcontractkit/chainlink-ccip/blob/30c877439f7a797edc0aaa6909433bb929082e10/shared/plugin_processor.go#L39-L44. Maybe `TokenDataHandler`?

Reply:
Or perhaps `TokenDataObserver` and the method is called `ObserveTokenData`

Reply:
`TokenDataObserver` sounds good to me